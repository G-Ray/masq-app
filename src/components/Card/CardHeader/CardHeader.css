.CardHeader {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.CardHeader .line {
  width: 40px;
  height: 4px;
  border-radius: 2px;
}
