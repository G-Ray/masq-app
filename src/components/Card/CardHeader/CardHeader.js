import React from 'react';

import SwitchButton from '../../SwitchButton/SwitchButton';

import './CardHeader.css';

// TODO: Add image in header
export default function CardHeader(props) {
  const { color, enabled } = props;
  const style = { backgroundColor: color };

  return (
    <div>
      <div className="CardHeader">
        <div className="line" style={style} />
        <SwitchButton color={color} checked={enabled} />
      </div>
    </div>
  );
}
