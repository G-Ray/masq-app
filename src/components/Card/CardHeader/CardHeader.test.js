import React from 'react'
import renderer from 'react-test-renderer'
import CardHeader from './CardHeader'

it('renders CardHeader disabled', () => {
  const component = renderer.create(<CardHeader enabled={false} color='blue' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders CardHeader enabled', () => {
  const component = renderer.create(<CardHeader enabled color='blue' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
