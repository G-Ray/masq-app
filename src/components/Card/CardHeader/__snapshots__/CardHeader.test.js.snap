// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardHeader disabled 1`] = `
<div
  className="CardHeader"
>
  <div
    className="line"
    style={
      Object {
        "backgroundColor": "var(--grey2-color)",
      }
    }
  />
  <div
    className="SwitchButton"
  >
    <label
      style={
        Object {
          "color": null,
        }
      }
    >
      DISABLED
    </label>
    <div
      className=""
      id="switch-container"
      onClick={[Function]}
    >
      <div
        className=""
        id="background"
        style={Object {}}
      />
      <div
        className=""
        id="cursor"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`renders CardHeader enabled 1`] = `
<div
  className="CardHeader"
>
  <div
    className="line"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="SwitchButton"
  >
    <label
      style={
        Object {
          "color": "blue",
        }
      }
    >
      ENABLED
    </label>
    <div
      className=""
      id="switch-container"
      onClick={[Function]}
    >
      <div
        className=""
        id="background"
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      />
      <div
        className=""
        id="cursor"
        style={
          Object {
            "transform": "translate3d(26px, 0, 0)",
          }
        }
      />
    </div>
  </div>
</div>
`;
